src/IdentityServer4/src/Configuration/CryptoHelper.cs

```diff
@@ -137,6 +137,12 @@ internal static bool IsValidCurveForAlgorithm(ECDsaSecurityKey key, string algor
 
             return true;
         }
+        internal static bool IsValidCrvValueForAlgorithm(string crv)
+        {
+            return crv == JsonWebKeyECTypes.P256 ||
+                   crv == JsonWebKeyECTypes.P384 ||
+                   crv == JsonWebKeyECTypes.P521;
```

Comment:
Maybe JsonWebKeyECTypes.P512 instead of JsonWebKeyECTypes.P521?

Reply:
Hi @vanbukin,

Good question, because RFC 7518 specify P256, P384 and P521 [RFC 7518 - Section 6.2.1.1](https://tools.ietf.org/html/rfc7518#section-6.2.1.1)

Reply:
This might be helpful for someone https://crypto.stackexchange.com/a/6221